wordpress-mobile/WordPress-iOS
# Tweak Jetpack Landing Screen prompts spacing and colors to align with the design

Fixes #19474 and #19489 

## Context
This PR tweaks the prompts text color and it reduces the prompts line spacing. The latter turned out to be a bit more challenging, and multiple approaches have been attempted before finding the one that works:
1. In `JetpackPromptView`, I tried using the `lineSpacing` modifier on `Text` but negative values have no effect.
2. I tried passing an `AttributedString` object to `Text`, but `ParagraphStyle` has no effect.
3. I tried embedding `UILabel` in `JetpackPromptView` and use `NSAttributedString` with `NSParagraphStyle.lineSpacing`. Same as 1st approach, setting negative `lineSpacing` values have no effect.
4. Similar to the 3rd approach, I used a `UITextView` this time, and negative `lineSpacing` values worked! But SwiftUI struggled to layout a list of `UITextView` and the end result looked like [this](https://user-images.githubusercontent.com/9609223/197193053-2be7e970-7cc0-48db-8250-87f503bcb6f1.png).

**Solution**
Since only the 4th approach gave some positive results, because at least negative `lineSpacing` values worked, I realized that the Landing Screen feature is using more UIKit than SwiftUI:
- The `InfiniteScrollerView` is written in UIKit.
- `JetpackPromptView` SwiftUI View was just wrapping a UIKit component.
- The only pure SwiftUI component was the `JetpackLandingScreenView`. Which is essentially a `VStack` of `JetpackPromptView`.

So the solution is to refactor `JetpackLandingScreenView` to UIKit and replace the `VStack` with a `UIStackView` embedding a list of `UITextView`. For each textView, we set an instance of `NSAttributedString` with the following attributes: Font, Foreground Color and ParagraphStyle ( lineSpacing property ).

https://user-images.githubusercontent.com/9609223/197194755-24361b55-3ef0-405c-8c93-a3a1044d9ff8.mp4

## Screenshots
| Before | iPhone | iPad |
| ------- | ------- | ---- |
| ![jp-iphone-before](https://user-images.githubusercontent.com/9609223/197196183-59feeb55-26fe-43fa-9f1c-758103a82c94.png) | ![](https://user-images.githubusercontent.com/9609223/197196522-85129c97-9bcc-4fa5-8ad2-e74f56997827.png) | ![ipad](https://user-images.githubusercontent.com/9609223/197197511-5658c310-2cfc-4769-a5ca-dd53f6d30a78.png) |

**Android previews for reference**

| Light | Dark |
| ----- | ----- |
| ![](https://user-images.githubusercontent.com/4588074/193870305-19fd60a0-7b2f-4d44-a5c8-fd571c1b0d76.png) | ![](https://user-images.githubusercontent.com/4588074/193870326-c6d78fc8-c177-4f86-b3f2-948f584d571a.png) |

## To test:
1. Clean install then log out to reach Landing Screen.
2. Verify the the spacing between prompts, the prompt's line spacing and color is visually close to Android screenshots.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
